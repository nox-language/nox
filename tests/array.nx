let type intArray = array of int
    var len := 12
    var a := intArray[len] of 0
    var b := intArray[12] of 7
    var value := a[0]
in (
    printi(value);
    a[0] := 10;
    printi(a[0]);
    for i := 0 to len - 1 do
        a[i] := i;
    for i := 0 to len - 1 do
        printi(a[i])
)
end
