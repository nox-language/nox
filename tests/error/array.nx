fun main() = (
    var a = [0; 12];
    var value = a[0];
)
