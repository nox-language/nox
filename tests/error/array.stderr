error: Undefined function `a`
   --> tests/error/array.nx:3:17
  |
3 |     var value = a[0];
  |                 ^ help: try the array indexing syntax a.[0]

